proknow-sdk/Patient/Entities/StructureSet/StructureSetDraftLockRenewer.cs

```diff
@@ -56,7 +59,15 @@ public void Start()
         /// </summary>
         public void Stop()
         {
-            _timer.Dispose();
+            // Wait for the timer to be disposed so that all Run callbacks, which are queued on another thread, have completed
+            using (var waitHandle = new ManualResetEvent(false))
+            {
+                // If this is the first time we've tried to dispose the timer, wait a few seconds for it to be disposed (it must wait for queued Run callbacks to finish)
+                if (_timer.Dispose(waitHandle) && !waitHandle.WaitOne(_timerDisposalTimeout))
+                {
+                    throw new TimeoutException("Timeout waiting for structure set draft lock renewer timer to stop.");
+                }
+            }

```

Comment:
I'm not sure I understand the `Dispose` method. From the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.timers.timer.dispose?view=netcore-3.1), it looks like it only takes a boolean. Is this an overload?

Reply:
Yes.  This particular overload will signal to the passed in waitHandle when the timer has actually been disposed.  The timer won't be disposed until all of the Run callbacks in the queue have been processed.  This guarantees that the Run method won't be called after we've disposed the timer and released the edit lock.